org.overte.Overte.yml

```diff
@@ -0,0 +1,88 @@
+app-id: org.overte.Overte
+runtime: org.kde.Platform
+runtime-version: '5.15-23.08'
+sdk: org.kde.Sdk
+# QtWebEngine is not part of the KDE runtime,
+# so we use this package, which is also maintained by KDE developers.
+base: io.qt.qtwebengine.BaseApp
+base-version: '5.15-23.08'
+command: Overte
+separate-locales: false
+rename-icon: Overte
+finish-args:
+  - --share=network
+  - --share=ipc # For Discord integration
+  - --device=dri # Graphics driver
+  - --socket=pulseaudio
+  # Overte requires OpenGL ES 3.2. While most Mesa drivers don't have this fully implemented,
+  # we try anyway, since it works on some devices.
+  - --env=MESA_GLES_VERSION_OVERRIDE=3.2
+  # QtWebEngine is shipped by the baseapp and not by the Sdk, which makes it use a non-standard path.
+  - --env=QTWEBENGINEPROCESS_PATH=/app/bin/QtWebEngineProcess
+  # KDE seems to do some aggressive theming which breaks things.
+  # See issue: https://github.com/overte-org/overte/issues/167
+  - --env=XDG_CURRENT_DESKTOP=foo
+  # While we technically support Wayland, there are issues with mouse capture: https://github.com/overte-org/overte/issues/628
+  - --socket=x11
+  - --env=QT_QPA_PLATFORM=xcb
+modules:
+  - name: unappimage
+    buildsystem: simple
+    build-commands:
+      - make -C squashfs-tools install INSTALL_DIR=/app/bin
+    sources:
+      - type: git
+        url: https://github.com/refi64/unappimage
+        commit: d7f86f2a0d7ec3a69211125207d5f127386b849a
+
+  - name: Overte
+    buildsystem: simple
+    cleanup:
+      - squashfs-root
+    sources:
+      - type: extra-data
```

Comment:
why is this extra-data?

Reply:
This is an AppImage being extracted. Is there a preferable method?
Maybe the extra-data is what calls the required apply_extra.sh script?

Reply:
> This is an AppImage being extracted. 

this bear no relation to extra-data

Reply:
why not just building from source?

Reply:
> why not just building from source?

Flathub doesn't allow an internet connection while building from source, which is required in this case.

Reply:
I just checked and it indeed is the reason for the script being called, which is needed to extract the AppImage.

> When installing the Flatpak package containing an extra data source, Flatpak calls the apply_extra script( …)

https://docs.flatpak.org/en/latest/module-sources.html

Is there any better way to run such operations while building the Flatpak?

Reply:
or use a standard archive format for the build artifacts. nonethess making your build process being able to work offline would improve on everything.

`extra-data` still isn't suited. This is reserved for license issue (ie when stuff isn't redistributable).

Reply:
> I just checked and it indeed is the reason for the script being called, which is needed to extract the AppImage.

Maybe I was unclear. The use of extra-data is irrelevant to extracting appimage, which is probably one of the worst way to provide the binaries here....

You should NOT use extra-data unless for licensing reasons.

Reply:
> I just checked and it indeed is the reason for the script being called, which is needed to extract the AppImage.

You missed the first line of extra-data which clearly states it is for avoiding license and redistribution clauses https://docs.flatpak.org/en/latest/module-sources.html#extra-data

> Extra data is often used for proprietary applications to avoid redistribution by downloading the binary package provided by the original distributors during installation by the user.

Reply:
Switched away from extra-data.